src/test/spec/retryable_writes/mod.rs

```diff
@@ -0,0 +1,374 @@
+mod test_file;
+
+use futures::stream::TryStreamExt;
+use semver::VersionReq;
+
+use test_file::{Result, TestFile};
+
+use crate::{
+    bson::{doc, Document},
+    concern::{Acknowledgment, ReadConcern, WriteConcern},
+    error::ErrorKind,
+    options::{ClientOptions, CollectionOptions, FindOptions, InsertManyOptions},
+    test::{assert_matches, run_spec_test, EventClient, TestClient, LOCK},
+};
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn run_spec_tests() {
+    async fn run_test(test_file: TestFile) {
+        for test_case in test_file.tests {
+            if test_case.operation.name == "bulkWrite" {
+                continue;
+            }
+
+            if test_case.description
+                == "InsertOne fails after multiple retryable writeConcernErrors"
+            {
```

Comment:
[This](https://github.com/mongodb/specifications/blob/192753e0ee9e7dd8ea30c10d055e119c7db1b99d/source/retryable-writes/tests/insertOne-serverErrors.json#L957) is the test that is causing issues with the error labels. The configured failpoint does not contain a label; however, the test results expect a RetryableWriteError label. This originally caused me to believe that the driver should be adding a RetryableWriteError label to write concern errors from 4.4+ servers, but Jeremy & Emily indicated that the server itself should be adding the label to the error from the configured failpoint on 4.4+ (more info [here](https://github.com/mongodb/mongo/wiki/The-%22failCommand%22-fail-point) under the "errorLabels" bullet point). However, it appears some servers are not doing this (verified in the shell and by printing out the error received from the server in the driver before driver label addition). We are currently investigating why this issue is occurring, but in the meantime I've skipped the test.